modifying vanilla price input and description. setting up american op…

esztermarton opened the pull request:
…tion input and description

editing vanilla prices to take in cost_of_carries and riskless_rate while keeping compatibility with version that takes in discount_factors. adding reference to american_options

adding skeleton for american pricing

call option written down

starting to write down put

small progress with put option

progress with put, refactoring to make slightly nicer

further progress on calculating s** crit for put

first attempt at tests - imports not working

importing option price from black scholes directly

renaming a few modeuls

adding numpy dep;

responding to PR comments

esztermarton committed:
modifying vanilla price input and description. setting up american option input and description

editing vanilla prices to take in cost_of_carries and riskless_rate while keeping compatibility with version that takes in discount_factors. adding reference to american_options

adding skeleton for american pricing

call option written down

starting to write down put

small progress with put option

progress with put, refactoring to make slightly nicer

further progress on calculating s** crit for put

first attempt at tests - imports not working

importing option price from black scholes directly

renaming a few modeuls

adding numpy dep;

responding to PR comments

test now failing on assertion error

adding more tests and correcting a typo

correct equation

all tests passing with outputs. will now look for bugs;

almost done

cyrilchim commented on tf_quant_finance/black_scholes/american_prices_test.py:
```diff
@@ -0,0 +1,54 @@
+          dtype=None))
+    expected_prices = np.array(
+        [0.03, 0.59, 3.52, 10.31, 20.0])
+    self.assertArrayNear(expected_prices, computed_prices, 1e-10)
```
You need to add 

if __name__ == '__main__':
  tf.test.main()

cyrilchimisov-ai committed:
Replace v1.while_loop with v2.while_loop and import tensorflow as tf with
import tensorflow.compat.v2 as tf. Cleanup lint errors.

PiperOrigin-RevId: 293820165

cyrilchimisov-ai committed:
Remove tf.debugging from hull_white module since dimension `dim` and batch dimensions are statically known.

PiperOrigin-RevId: 293836469

cyrilchimisov-ai committed:
Change `import tensorlow as tf` to `import tensorflow.compat.v2 as tf` in experimental module

PiperOrigin-RevId: 293844965
